### PR TITLE
Add version requirement for filelock and nni

### PR DESCRIPTION
**Pull request by KumoLiu**
Add version requirement for filelock and nni

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by ericspod** [comment]
We may want to change how cv2 is imported in the `video_dataset.py` file which is a bit of an usual workaround. It currently will import it as an option but used everywhere without checking if it's present, so having regular import statements in the methods which use it may be a better idea and get around the issue of needing cv2 to generate docs.

**Comment by KumoLiu** [comment]
> We may want to change how cv2 is imported in the `video_dataset.py` file which is a bit of an usual workaround. It currently will import it as an option but used everywhere without checking if it's present, so having regular import statements in the methods which use it may be a better idea and get around the issue of needing cv2 to generate docs.

It's wired, opencv-python-headless is added in this PR: https://github.com/Project-MONAI/MONAI/commit/12684884a93173bed23190ff7a3d0e674a3e6a9f.
Before we don't require this package to be installed.

File: docs/requirements.txt [hunk 1]
```diff
@@ -35,7 +35,6 @@ pydicom
 h5py
 nni; platform_system == "Linux"
 optuna
-opencv-python-headless
```
[the remark this answers is not in the document]
Comment (ericspod) [reply docs/requirements.txt:38]: Ok that's fine for the Dockerfile. The issue here I think with the build-docs fail is that it's removed from this requirements file which is for sphinx?

[the remark this answers is not in the document]
Comment (KumoLiu) [reply docs/requirements.txt]: Oh yes, thanks for pointing it out!

**Review by ericspod — APPROVED** [review]
Looks good now for these changes.

**Pull request merged**